api/src/services/collections.ts

```diff
@@ -204,12 +204,11 @@ export class CollectionsService {
 				await clearSystemCache({ autoPurgeCache: opts?.autoPurgeCache });
 			}
 
-			// Refresh the schema for subsequent reads
-			this.schema = await getSchema();
-
 			if (opts?.emitEvents !== false && nestedActionEvents.length > 0) {
+				const updatedSchema = await getSchema();
+
 				for (const nestedActionEvent of nestedActionEvents) {
-					nestedActionEvent.context.schema = this.schema;
+					nestedActionEvent.context.schema = updatedSchema;
```

Comment:
We should probably revert both updates of the schema that were implemented in https://github.com/directus/directus/pull/23888/files and add the skip flag in the second place as well.

Reply:
Ahhhh yes there's another one too. Thanks!

Reply:
The skip flag is only added in `readMany`.

Reply:
Refactored in https://github.com/directus/directus/pull/23991/commits/4773da9c0d054ca8e4327d19d883861be82c00c6

Reply:
Oh yeah, `validateAccess` is only called in `readMany`